lib/rules/no-array-constructor.js

```diff
@@ -107,6 +133,17 @@ module.exports = {
 				let fixText;
 				let messageId;
 
+				const nonSpreadCount = node.arguments.reduce(
+					(count, arg) =>
+						arg.type !== "SpreadElement" ? count + 1 : count,
+					0,
+				);
+
+				const shouldSuggest =
+					node.parent.type === "ChainExpression" ||
```

Comment:
```suggestion
					node.optional ||
```

For example, in `Array?.(0, 1, 2).forEach(doSomething);`, direct parent of `Array?.(0, 1, 2)` is not `ChainExpression`.